src/Adapters/Files/FilesAdapter.js

```diff
@@ -56,6 +58,30 @@ export class FilesAdapter {
    * @return {string} Absolute URL
    */
   getFileLocation(config: Config, filename: string): string {}
+
+  /**
+   *
+   * @param {string} filename
+   *
+   * @returns {null|*|Parse.Error} null if there are no errors
+   */
+  validateFilename(filename): ?Parse.Error {}
+}
+
+export function validateFilename(filename): ?Parse.Error {
```

Comment:
Really wanted to change this into an Abstract Class with a default implementation for validateFilename, but couldn't quite grok how to do that in the ES5/6 world.    Suggestions are welcome.